src/iftraf.c

```diff
@@ -1,1116 +1,1131 @@
-                        NETIF_MAX_PKT_BURST, NULL);
-        }
-		
-        RTE_LOG(DEBUG, IFTRAF,
-            "%s: iftraf ring[%d] free [%d] pkts\n",
-            __func__, cid, count);
-
-        count = 0;
-		
-    }	
-
-    /* free iftraf ring */
-    iftraf_ring_free();
-
-    /* free tlb */
-    if (iftraf_tbl) {
-        count = 0;
-        /* delete and free all entry added to iftraf_tbl */
-        for(hash = 0; hash < IFTRAF_TBL_SIZE; hash++) {
-            list_for_each_entry_safe(entry, nxt, &iftraf_tbl[hash], list) {
-                list_del(&entry->list);
-                rte_free(entry);
-                count++;
-            }
-        }
-
-        RTE_LOG(DEBUG, IFTRAF,
-            "%s: iftraf_tbl free [%d]\n",
-            __func__, count);
-
-        rte_free(iftraf_tbl);
-    }
-
-    /* free tlb */
-    if (iftraf_iftbl) {
-        count = 0;
-        /* delete and free all entry added to iftraf_tbl */
-        for(hash = 0; hash < IFTRAF_IFTBL_SIZE; hash++) {
-            list_for_each_entry_safe(entry, nxt, &iftraf_iftbl[hash], list) {
-                list_del(&entry->list);
-                rte_free(entry);
-                count++;
-            }
-        }
-		
-        RTE_LOG(DEBUG, IFTRAF,
-            "%s: iftraf_iftbl free [%d]\n",
-            __func__, count);
-
-        rte_free(iftraf_iftbl);
-    }
-    
-    iftraf_variable_reset();
-    RTE_LOG(INFO, IFTRAF,
-        "%s: %s\n", __func__, "iftraf disabled");
-
-    return EDPVS_OK;
-}
-
-static void iftraf_sorted_list_init(void) {  
-    portid_t devid = 0;
-    sorted_list_initialise(&iftraf_sorted_list);
-
-    for (devid = 0; devid < NETIF_MAX_PORTS; devid++) {
-        sorted_list_initialise(&sorted_list[devid]);
-    }
-}
-
-static int iftraf_sockopt_set(sockoptid_t opt, const void *conf, size_t size)
-{
-     switch (opt) {
-     case SOCKOPT_SET_IFTRAF_ADD:
-          return iftraf_enable_func(); 
-     case SOCKOPT_SET_IFTRAF_DEL:
-          return iftraf_disable_func(); 
-
-     default:
-          return EDPVS_NOTSUPP;
-     }
-}
-
-
-static struct dpvs_sockopts iftraf_sockopts = {
-    .version        = SOCKOPT_VERSION,
-    .set_opt_min    = SOCKOPT_SET_IFTRAF_ADD,
-    .set_opt_max    = SOCKOPT_SET_IFTRAF_DEL,
-    .set            = iftraf_sockopt_set,
-    .get_opt_min    = SOCKOPT_GET_IFTRAF_SHOW,
-    .get_opt_max    = SOCKOPT_GET_IFTRAF_SHOW,
-    .get            = iftraf_sockopt_get,
-};
-
-int iftraf_init(void)
-{
-    int err;
-	
-    iftraf_disable = true;
-
-    iftraf_tlb_rnd = (uint32_t)random();
-
-    iftraf_sorted_list_init();
-
-    if ((err = sockopt_register(&iftraf_sockopts)) != EDPVS_OK)
-        return err;
-
-    return EDPVS_OK;
-}
-
-int iftraf_term(void)
-{
-    int err;
-
-    err = sockopt_unregister(&iftraf_sockopts);
-    if (err != EDPVS_OK)
-        return err;
-
-    err = iftraf_disable_func();
-    if (err != EDPVS_OK)
-        return err;
-
-    return EDPVS_OK;
-}
+#include <stdio.h>
```

Comment:
add copy right
/*
 * DPVS is a software load balancer (Virtual Server) based on DPDK.
 *
 * Copyright (C) 2017 iQIYI (www.iqiyi.com).
 * All Rights Reserved.
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 */